schemas: iio: iio-consumer: add io-backends schema

nunojsa opened the pull request:
IIO backends are devices connected to another IIO devices (consumers of the backends) providing services so that the consumer IIO device can be fully functional. One typical example of such an Hardware arrangement would be an high speed ADC that is connected to an FPGA IP core (through a serial data interface as LVDS) capable of handling the high sample rates. Examples of the provided services would be channel enablement/disablement.

While at it, fixed a typo in the top level description 'thes -> they'.

---

For more information, there's the current linux [patchseries](https://lore.kernel.org/linux-iio/20231208-dev-iio-backend-v2-0-5450951895e1@analog.com/) adding support for iio backends.

robherring commented:
Actually, I have further comments on this. Will comment on the patch series instead.

nunojsa commented on dtschema/schemas/iio/iio-consumer.yaml:
```diff
@@ -30,7 +31,19 @@ properties:
 
+  io-backends:
+    $ref: /schemas/types.yaml#/definitions/phandle-array
+    description: >
```
ack

nunojsa commented on dtschema/schemas/iio/iio-consumer.yaml:
```diff
@@ -30,7 +31,19 @@ properties:
+
 dependencies:
   io-channel-names: [io-channels]
+  io-backends-names: [io-backends]
```
ack

nunojsa commented:
> Actually, I have further comments on this. Will comment on the patch series instead.

Alright, thanks for the feedback... I'll reply in the series

nunojsa commented:
@robherring, one question... Digging though git log I ended up in here:

https://lore.kernel.org/all/20230507171219.232216-1-krzk@kernel.org/

Is this the preferred way? When things settle, should I send a patch to the mailing lists? I guess this way is better to get the proper acks...

robherring commented:
> Is this the preferred way? When things settle, should I send a patch to the mailing lists? I guess this way is better to get the proper acks...

Either way is fine. For the mailing list though, use devicetree-spec or I'll probably miss it.

nunojsa commented:
Pull updated... v2:

* Added 'io-backend-cells';
* Removed > in io-backends description;
* io-backends-names -> io-backend-names;
* Added acked-by tags.

I've set the cells as an enum even tough we don't really have a usecase for it. I'm also adding the cells as suggested by you as it might be needed in the future and at the very least it's good to identify backend providers. Though, I can change the schema to `const: 0` if you prefer.

Links for the acks:
 * https://lore.kernel.org/linux-iio/4b1ffdc4-edce-4a69-a30b-45c29741dc2c@foss.st.com/
 * https://lore.kernel.org/linux-iio/a0e01566ef2317ba35931526c4496723839be5c2.camel@gmail.com/T/#m235df32948171a3646c5a03d246b53e21c2ed6ad
